### ISSUE TITLE
Create a DMG package for Spyder

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#697]
_From [mjmare31...@gmail.com](https://code.google.com/u/114349152275557912536/) on 2011-06-24T14:42:33Z_

It would be very nice to have a DMG package of Spyder. Macports installs far to much crap.

An (working) easy_install would be nice as well.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=697_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-06-25T05:03:48Z_

Regarding the "working" easy_install, this is beyond our scope as the "non-working" part concerns the PyQt library.

As for the dmg package, why not?
But we need a contributor who is interested/working with MacOS X... none of the current developers/contributors are (as far as I know)

**Status:** ContributorNeeded  
  **Labels:** -Type-Defect Type-Task OpSys-OSX

**Comment by spyder-issues-migrator** [comment]
_From [fhe...@gmail.com](https://code.google.com/u/111731474599148743921/) on 2011-11-02T11:00:06Z_

As I just got a Mac and am also interested in changing from Matlab to scipy, I am most interested in helping. 

How do I start?

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-11-26T07:47:54Z_

Any Mac user may help him getting started? Carlos?

**Cc:** ccordoba12

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2011-11-26T15:22:56Z_

I think the right way to do it would be to use pyinstaller or py2app. I tried to use pyinstaller about 2 months ago but was incapable to produce a Mac app. However that was before you added the changes to generate a Windows self-executable.

Could you tell me how you use py2exe? Just give a few simple steps, maybe I can reproduce them in Mac with py2app, who knows?

It would be great to have a dev with in-depth knowledge of OSX. I'm just a causal user because I'm all the time in Linux but I'll try to do my best to generate an app.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-07T14:53:11Z_

I'm also trying to produce a Mac package. The question is exactly what the result should be. Here are some options:
* A package which just unpacks spyder to your current python site-packages but includes no dependencies, you would have to install pyside etc from their own install packages

* A multi-package installer, which automatically installs python, pyside/QT, possibly numpy and pylab from their respective package installers, all to the site-packages folder - you don't get a .app, but maybe a shortcut to launch spyder

* A .app mac application which packs up pyside/QT etc into a drag-and-drop executable.
However, in this case, should the packaged dependencies (pyside etc) also be included in the runtime path when you run spyder? 
Should spyder include its own internal python runtime and everything packed into one place?

Basically, the question is whether spyder should be running as an app seperate from the python installation and all the libs you want installed to run your scripts. Or should it be just another lib installed along with the rest of your site-packages?

Feedback welcome, i'm just now getting to grips with packaging/distribution and py2app

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-02-08T07:01:42Z_

Hi Jon, thanks a lot for coming to help us with this one. It'd be really great if we could support Mac with an in-house App.

About your question: I think we just need to install Spyder as a lib, along with the rest of the user site-packages in the default Mac python installation. The thing is that Nokia provides easy-to-use Mac installers for Qt and PySide so we shouldn't worry about packaging our own versions of them.

If the user has EPD, homebrew or his own virtualenv, he can easily tell Spyder to use any of these environments just pointing to its main python executable in:

Tools > Preferences > Console > Advanced Settings > Python Executable

If you need any additional help, please don't hesitate to tell us!

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-08T08:25:55Z_

Ok, here's my thoughts:

If we just want spyder in the libs and expect people to install pyside from nokia (which is MUCH easier than installing PyQT!) then this is more a question of updating the install instructions for MacOSX. 
Spyder can be installed from source quite easily, but for extra simplicity I could try making a .pkg (NOT a .app) which will basically just run setup.py install but wrapped in a nicer interface and possibly create a launcher shortcut.

For extra ease of user it could be possible to make an mpkg which includes pyside and qt from nokia with the spyder pkg and installs them at the same time.

This is different from having a spyder.app, which would be a more like standalone IDE like matlab or pydev which would run your scripts in the interpreter you specified in preferences. I think this would be the nicest (you just run spyder and go!) but turning python into .apps seems to be very flaky.

For now i will try and get an installable spyder.pkg which will install to your default site-packages and require you to install pyside and qt seperately. Still learning how to create installers for mac - seems more of a pain as soon as you're not using xcode/objc which is not really surprising for apple.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-08T17:35:31Z_

Ok, so I was able to use the bdist_mpkg tool to create a .pkg installer. This has been tested on OSX10.7 only - because there tends to be small changes in the install locations between versions of OSX it's not guaranteed to work anywhere else.

It installs Spyder to
/Library/Python/2.7/site-packages
and places an executable in 
/usr/local/bin/ (so its accessible from the terminal)

If someone could test this, it would be great. Here is the zipped spyder .pkg: http://dl.dropbox.com/u/6613592/spyder-2.1.7-py2.7-macosx10.7.mpkg.zip And I also created a DMG which includes installation instructions and links to PySide and QT. http://dl.dropbox.com/u/6613592/Spyder.dmg If there's some interest it would be easy to include those installers in the DMG, but for now its a bit redundant.

Note that spyder will be available to any python 2.7 that's installed (as long as you're not using virtualenv) - i can import it from both Apple python and EPD python. It seems /Library/Python/2.7/site-packages are "global" while those installed for e.g in /Library/Frameworks/Python.framework/Versions/... are not.

The next step would be to make this installer more flexible so it can be used with OSX10.6 also (I'm not planning on supporting anything \<10.6 because of 32/64 bit issues). I'm still figuring out mac packaging, if anyone has good resources that would be handy, for example to add scripts to install to the right place depending on environment vars etc.

Anyway, hope this package is useful!

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-02-08T20:48:28Z_

Great work Jon! It would certainly be really useful to all our Mac users out there.

I could help you testing but I have OSX10.6 (which comes with Python 2.6). Will it work here? or do I have to wait for the improvements you mention?

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-09T01:10:22Z_

@-ccordoba12 - if you install with Python 2.6 it won't work normally, however i think if you install python 2.7 from python.org it should, maybe you could test this? I'm still trying to work out exactly where OSX gets all its python bits and where this is configured, there seems to be at least 3 places where python packages can live... if you have any insights into this it would be helpful.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-02-09T08:34:28Z_

Ok, I'll try to do it tomorrow (today is impossible for me).

Where does the installer from python.org put its files? In /Library/Frameworks/Python.framework/Versions or in /Library/Python/? I just don't want to screw up my Python Mac.

You can also take a look at PyInstaller, which I've heard many people use to create Mac Apps (I tried to use it for a couple of hours and didn't get too far...)

Thanks for all your efforts. Every step in this direction is a terrific improvement for the project.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-02-19T15:47:14Z_

@-ccordoba12 - sorry i've not been on this for a while, been busy. I was having another look at packaging and trying to get some expert help (though not had much luck getting replies yet).

Anyway, a couple of updates:
Building packages for several versions of OSX is a bit of a black art it seems, there's not a lot of clear information on how to do it. But i think i've got the general idea and should be able to make a package for osx10.6 with python 2.6 (actually i think it would be for 2.6 in general, no matter the machine)

BUT I need YOUR help! Can you tell me what the contents of the folder "/Library/Python" is on your 10.6 machine? Note that's the library folder under Macintosh HD, not in your home folder.

As for creating an App, its not as straightforward as you might think. Mac apps are self contained - they have everything inside one bundle. The problem is that for spyder that means pulling in Qt and everything, probably even the python runtime. This is hard to get right. But even worse, it's not actually what most people will want! If you're doing scientific python its likely that you want spyder running in the same python install as all the libraries you installed.

That's not to say its impossible, just that its a lot more work than creating a package. All the package file does is offer an alternative to installing from source using python setup.py install and installing all the dependancies manually.

Personally I don't have much time, so what i'd like to do is to simply make it easier to install Spyder and its dependencies. I think a realistic and acheivable target is a combined package which would install PySide+Qt+Spyder in one go into your current python install.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-02-19T21:07:39Z_

Jon, I'm really sorry with you for letting you waiting. I've been using the little Spyder time I have to try to improve the unsatisfactory situation for automatic quote insertion I left in version 2.1.7 (see issue #872 for all the things I have to do).

I totally agree with you. Spyder by itself is not of much value. It has to be used with your preferred scientific libraries to get the maximum advantage out of it. So I'm perfectly fine with a package instead of an App.

Without installing the python DMG from python.org, what I have in /Library/Python is this:

$ls /Library/Python/
2.3 2.5 2.6

Inside every version I have just one dir: 'site-packages' and inside it a README which has this text:

"This directory exists so that 3rd party packages can be installed
here. Read the source for site.py for more details"

In the 2.6 dir I have two packages: IPython and pip, which I guess I installed using the system-wide python before installing Homebrew.

So your plan is to put there PySide and Spyder? Awesome! I didn't know about it. If you need anymore help, please let me know and give a little kick if I'm late in answering back.

Cheers,
Carlos

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2012-03-18T14:16:21Z_

**Status:** HelpNeeded

**Comment by spyder-issues-migrator** [comment]
_From [ebori...@gmail.com](https://code.google.com/u/102672023975932251472/) on 2012-03-19T09:32:12Z_

After you've gone through all of this, is it really better than having MacPorts do it for you?

Yes, it installs a lot to "just" get Spyder going, but Spyder is a development environment with *lots* of things you either want or need to have installed to make it useful: python, Qt4, numpy, scipy, matplotlib, pyqt4, pil, h5py....  All in the right places and happy with each other. Of that list, only python and numpy are installed on a fresh OSX install, so everything else needs to come from somewhere. (And a number of the others have dependencies that are also not native on the mac, like hdf5 for h5py.)

You can cut down on the things installed by 'port install py27-spyder' by first installing 'py27-matplotlib +qt4-tkinter' (this avoids all the X11 things that get pulled in for just to support py27-tkinter)

When you are done, you now have MacPorts behind the scenes to handle updates to all those packages needed just to get Spyder up and running...

If you're installing on x86_64, non-universal, snow leopard (very specific for now; they are slowly adding more flavors) many of the packages get downloaded in a compiled form now to make the initial install much faster.

Just my 2c. (I am the port maintainer for this, so I am a little biased. ;)

**Comment by spyder-issues-migrator** [comment]
_From [adam.lw....@gmail.com](https://code.google.com/u/107757689056514629413/) on 2012-03-20T22:39:35Z_

IMO I think you're going about this the wrong way.

What I want is a self-contained .app that I can just run to get the IDE.
I have multiple python installations using pythonbrew and virtualenv. I don't want spyder installed into my OS-X python install, I won't want to have to install it into N python installations, I don't want to maintain a python installation just for Spyder.
I just want a self contained .app, that I can run and be done with.

I've spent the day looking for one and the only ones available are non-free. I think im going to have to go back to VI + console debugging. =(

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-03-21T03:55:12Z_

@-adam - I too wanted a Spyder.app, but when i looked into creating it I ran into a lot of difficulties.

The problem is that for spyder to be useful you need to be able to run python programs. Spyder tries to help by importing matplotlib (which in turn depends on numpy). So to get spyder working you need matplotlib and numpy into the same python interpreter used by sypder. Because spyder is written in python, you need AT LEAST one python installation to run spyder, and that installation needs to have matplotib and number installed.

This creates a problem, because on the one hand you have spyder which needs these things. On the other hand you want to write programs which probably also need numpy and matplotlib, plus all the other modules you need for your specific project.

So if you want to create a completely standalone spyder.app, you need to include spyder's dependancies in the app. Then if you want to write python programs in spyder, you want those programs to be able to access third party modules.


But you probably don't want to install those modules into the spyder.app own package, because they might mess up spyder itself. So they will in any case need to be installed into another python "installation" (either a whole other installation or a virtualenv). As you can see this is already getting complicated. But it gets worse. What do you do about matplotlib? Do you install it twice - once for spyder and once for your scripts? Do you make both environments share the same module?

I'm not saying any of this is impossible, it's just that to create a spyder.app means making a lot of opinionated decisions about how it should all work, and then actually making it work. It's not as easy as just using py2app - trust me, i tried!

That's not to say this isn't a worthy project - I for one would love to see spyder as a Matlab replacement where you just run a .app and everything works.

So I think the first thing to do down this route is to create a spec (mac specific for now) detailing exactly how it should behave - including what interpreted sypder should use, what interpreter scripts written in spyder should use, where the site-packages should be stored for both those interpreters, and so on.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-03-21T04:09:55Z_

@-eborisch

I'm not claiming it's "better" in an absolute sense - but it is better for certain people's way of working (mine included). 

All the .pkg does is make it easier to install spyder into your osx-python installation. It means you don't need to install another version of python through macports. Also, most of the dependencies are available as click-to-install packages - you can just download numpy, scipy, matplotlib, pyside/qt, and spyder and install them without ever seeing the command line. I find this is more user friendly.

Conversely I find installing through macports very time-consuming and I've had a generally bad experiences with it. Now this is just me, other people have different preferences, but i think enough people would rather not use macports to justify having a pkg installer.

Furthermore, figuring out how to package spyder is a first step to figuring out how it could be turned into a standalone app for super user friendly scientific programming matlab-style.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-03-21T07:21:01Z_

@-Adam: You don't certainly need to do install Spyder for every interpreter that you have. Right now it has support to deal with different interpreters/virtualenvs because you can change your main interpreter in

Tools > Preferences > Console > Advanced Settings > Python Executable

as I said in comment `#6`. I plan to improve it, using there the first interpreter in your path, saving the list of interpreters that you have been used and perhaps offering a list of your virtualenvs. But that's why Jon's work is really important: because it offers a way to do an easy system-wide installation that you can use with the interpreter you prefer.

@-Jon: Spyder doesn't need numpy/matplotlib or any other scientific package to *run*. They are just used to provide better facilities to the console but perhaps they're grabbed by py2app because we have some matplotlib patches on the tree. Instead it would be better to add rope, pyflakes, pep8 and pylint to the pkg (if they're not there already :-)

@-eborisch: Thanks a lot for your efforts supporting Spyder on MacPorts. We really appreciate it as we appreciate your patches too :) But Jon efforts have a lot of value for us because a lot of people use EPD on Mac and they just want an easy way to use Spyder with it.

**Labels:** MS-v2.2

**Comment by spyder-issues-migrator** [comment]
_From [nhoening](https://code.google.com/u/nhoening/) on 2012-03-29T05:02:23Z_

Hello, I (on OSX 10.6) decided to try this out.

I installed Python from http://python.org/download/ , and Qt and pyside as the README suggests.
However, when selecting the drive to install spyder to, it says:
"spyder requires Apple Python 2.7 to install"
and doesn't allow me to continue.

I am not sure I understand what "Apple Python" is.

I do have a Python2.7 in /Library/Python:
$ ls /Library/Python/
2.3	2.5	2.6	2.7

And also in /Library/Frameworks:

$ ls /Library/Frameworks/Python.framework/Versions/
2.7	Current
$ /Library/Frameworks/Python.framework/Versions/Current/bin/py 
pydoc             python            python-config     python2.7-32      pythonw           pythonw2.7
pydoc2.7          python-32         python2.7         python2.7-config  pythonw-32        pythonw2.7-32


So what excatly is expected here? 
Thanks for doing this,

-Nic

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-04-07T08:38:49Z_

@-Nic
Its seems that when i used the bdist_mpkg command it automatically creates a requirement for a the specific version of python that i have installed - that is, the apple-provided python installation that comes with OSX 10.7.

In any case, this installed won't work for python 2.6 because it installs to /Library/Python/2.7/site-packages

I'm still trying to find out how I can first detect the python version and then make the installer put things in the right place.

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-04-08T08:46:15Z_

OK so i'm trying a different method to manually create packages which simply copy spyderlib and spyderplugins to /Library/Python/2.7/site-packages (for python 2.7) 
or /Library/Python/2.6/site-packages (for python 2.6). The spyder executable which lets you launch spyder from the command line is copied to usr/local/bin in both cases

You can download these here, let me know if they work.
Python 2.7: http://dl.dropbox.com/u/6613592/spyder-2.1.9-py2.7.pkg Python 2.6 http://dl.dropbox.com/u/6613592/spyder-2.1.9-py2.6.pkg This version does not try to check that you have any of the requirements. Maybe it would be possible to add a check for PySide/PyQT, as well as to combine the installers into one which checks for the python version before installing. I don't have the knowledge to do that though. 

On the other hand i spent some time learning about python packaging and i have a much better idea of how it works, so i might be able to get something more interesting worked out, maybe even a .app after all!

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-04-08T09:00:22Z_

As an Easter egg (since it's easter) I also packaged a special version which includes my new icon them (which looks great on Mac!). It's based on 2.2dev with just the icons changed, enjoy! http://dl.dropbox.com/u/6613592/spyder-2.x-py2.7.pkg

**Comment by spyder-issues-migrator** [comment]
_From [AuburnAa...@gmail.com](https://code.google.com/u/101581579241260311066/) on 2012-07-07T17:05:31Z_

Hi all,

I'm a scientific programmer deciding if I want to switch from matlab to python and use spider.  I installed it on my home desktop, running Lion, through macports which was a little painful.  It seems like macports installed thousands of folders, seems kinda messy.  In any case, it works and I like spider very much.  

Here is my problem: macports is blocked at my work (along with dropbox and many other similar sights).  I can always download the necessary files at home and bring them to work for an install, but if macports is really critical I am screwed.  Any thoughts on how to fully bypass macports?

Thanks,

A

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-07-07T19:29:16Z_

I can't promise anything definitive, but during the next couple of weeks I'll have access to a Mac, so I'll try to create a real DMG (which I think is the best we can do for our Mac users). It doesn't seem that hard, but we'll see how it goes.

**Comment by spyder-issues-migrator** [comment]
_From [ebori...@gmail.com](https://code.google.com/u/102672023975932251472/) on 2012-07-07T19:35:34Z_

You can use 'sudo port mpkg py27-spyder' on your machine with macports to make a (large, as it includes all deps as well) installer package...

**Comment by spyder-issues-migrator** [comment]
_From [AuburnAa...@gmail.com](https://code.google.com/u/101581579241260311066/) on 2012-07-08T15:09:55Z_

Thanks e, I didn't realize the .mpkg that macports makes has everything in it.  I made a .pkg with macports, but that obviously doesn't have everything.  Kinda feel like an idiot now.  Anyway, I'll try the .mpkg tomorrow, hopefully it works!

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-07-17T23:05:24Z_

Hey guys, I just want to let you know that after a couple of weeks of hard work, I've created a self contained and full functional Spyder Mac app. I'll upload it to our Downloads section during the next few days.

**Status:** Started  
  **Owner:** ccordoba12  
  **Cc:** -ccordoba12

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-07-19T12:35:04Z_

Hey, that's great! How did you go about it? I was looking into this and got some way using py2app but hit a wall because of a bunch of import errors (and then had no more time :P) Are there source code changes too?

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-07-21T10:07:06Z_

Hi Jon, I had some troubles too. The hardest one was to notice that py2app compresses site-packages into a zip file (to save some space) and that spyder couldn't start this way because it can't load its icons.

After that everything was relatively easy. The other main trouble was to add all the std library modules to the app, because the user has to have them to get its job done :-) (py2app just add the ones needed by the program).

I made a script to create the app. I'll try to push my commits this afternoon and also the DMG.

I'll probably need your help to create the app in Lion, in case my app (created in Snow Leopard) doesn't work there.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-07-21T13:13:02Z_

Jon, I just uploaded the new DMG. Please help us to test it in Lion to see if it's working well.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-07-21T13:14:18Z_

This issue was updated by revision 19ca66588c04 .


- Besides the create_app.py script, I'm also adding an icns icon to go with the
app.

**Status:** Fixed

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-07-21T15:34:53Z_

Ok, so it loads fine on Lion. However, trying to use the interactive console to do from pylab import * fails, trying to run a script using the run button also fails.

\>>> Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 455, in icon
    project = self.treeview.get_source_project(fname)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/projectexplorer.py", line 664, in get_source_project
    return self.workspace.get_source_project(fname)
AttributeError: 'ExplorerTreeWidget' object has no attribute 'workspace'
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyde...

**Comment by spyder-issues-migrator** [comment]
_From [jon.cham...@gmail.com](https://code.google.com/u/104556211888717111926/) on 2012-07-21T15:34:53Z_

...rlib/plugins/editor.py", line 1865, in run_file
    self.configdialog_size = dialog.get_window_size()
RuntimeError: wrapped C/C++ object of type RunConfigOneDialog has been deleted

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-07-21T17:03:51Z_

Ok, thanks for testing. I'm really glad to hear the app is working in Lion.

Now the errors:

1. I think the pylab error is not entirely our fault, but a missing config in matplotlib. I'll double check it though to be sure.

2. The run dialog is an error I think I introduced a couple of commits ago.

I also discovered more errors when the user first starts the app (because of error `#2` and some missing stuff).

I'll try to fix them quickly and upload a new image ASAP.

**Status:** Started

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-07-22T15:15:44Z_

Jon, I uploaded a new image. I fixed error `#2` and the others I mentioned.

Error `#1` could be caused because you have a .matplotlib/matplotlibrc in your home directory without this line:

interactive: True

It seems to be required for new versions (probably >= 1.1)

**Comment by spyder-issues-migrator** [comment]
_From [paulrco...@gmail.com](https://code.google.com/u/113519093809978567556/) on 2012-07-22T17:22:35Z_

Folks, I just bought a new MacBook Pro and downloaded spyder from macports. It worked flawlessly and I was able to run it from the terminal window, but I couldn't find an icon to launch it from my applications folder.  So I uninstalled the macports version and downloaded the dmg from code.google.com/p/spyderlib/downloads.  It installs beautifully and the only trouble I'm having now is that my python is buggy!  Thanks, --Paul

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2012-07-22T19:00:12Z_

Paul, what do you exactly mean by "my python is buggy"? what error are you seeing?

**Comment by spyder-issues-migrator** [comment]
_From [nhoening](https://code.google.com/u/nhoening/) on 2012-07-23T16:11:56Z_

I just want to report that I tried it just now on my MacBook with Snow Leopard and it worked like a breeze. MacPorts installations take very long on this old thing, but this install was very fast. Many thanks to ccordoba12 for the hard work. I think this really increases the value of Spyder!

I did run into issue `#743` though (Plotting is not working in the console for Ipython > 0.11), but that is a general issue, of course (and the Python console does just fine).

**Comment by spyder-issues-migrator** [comment]
_From [ebori...@gmail.com](https://code.google.com/u/102672023975932251472/) on 2012-07-23T21:58:50Z_

FWIW, if you are set up with the defaults (not +universal, normal /opt/local/ and /Applications/MacPorts config for MacPorts, see [1]) most of the packages required for Spyder on SL will be downloaded compiled from packages.macports.com and installed without compilation. I did a full install of py27-spyder from an empty macports default install (so it would use binaries) at it took ~12 minutes, where a lot of that was download time (Qt4 is big in any form.)

I used:
\# port install py27-ipython +pyqt4
\# port install py27-matplotlib -tkinter+qt4  (to avoid all the stuff that gets pulled in to support tkinter)
\# port install py27-spyder

IPython inline plotting is working just fine for me (with IPython 0.13, also via MacPorts...)

[1] http://lists.macosforge.org/pipermail/macports-users/2012-June/029263.html for the full list of requirements for MacPorts to use binary packages when available.

**Comment by spyder-issues-migrator** [comment]
_From [paulrco...@gmail.com](https://code.google.com/u/113519093809978567556/) on 2012-07-23T22:09:21Z_

To ccordoba:

My apologies, I should have said "my python programming is buggy."  I'm new to the language (I'm an old lisp guy) and this forum.  Nought wrong with Spyder, as far as I know, just the guy who's using it.  

Thanks, --Paul

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-01-26T19:29:55Z_

Ok, I consider this issue fixed now. If you have problems with our DMG, please report them as separate issues.

I already have fixes for the most serious ones ( issue #1226 and issue #1186 ) that I plan to upload tomorrow. Thanks for your patience and I hope you enjoy our new and shiny app!

**Summary:** Create a DMG package for Spyder  \<span class="oldvalue"> (was: DMG package for Spyder) \</span>  
  **Status:** Fixed

**Comment by spyder-issues-migrator** [comment]
_From [eagled...@gmail.com](https://code.google.com/u/113243112442161874462/) on 2013-02-07T11:31:16Z_

For the "RuntimeError: wrapped C/C++ object of type RunConfigOneDialog has been deleted" error for me was caused by using an older version of Spyder (2.1.11) and trying to do certain things with matplotlib (usually subplot related). After upgrading to 2.1.13, I'm no longer getting this error and plots are working fine. If anyone else is having this issue try that.

For the record, I am using pythonxy 2.7.3.1, with upgraded PyQt (4.9.6).

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-04-06T20:20:23Z_

**Labels:** Cat-SpyderGUI
